scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_model.py

```diff
@@ -35,21 +38,27 @@ def plot(self, workspace_list, title):
             return
 
         if self.plot_figure:
-            self.plot_figure.clear()
             self.plot_figure = plot(workspaces, spectrum_nums=[1], fig=self.plot_figure, window_title=title,
-                                    plot_kwargs={'distribution': True}, errors=True)
+                                    plot_kwargs={'distribution': True, 'autoscale_on_update': False}, errors=True)
         else:
-            self.plot_figure = plot(workspaces, spectrum_nums=[1], window_title=title, plot_kwargs={'distribution': True},
+            self.plot_figure = plot(workspaces, spectrum_nums=[1], window_title=title, plot_kwargs={'distribution': True,
+                                                                                                    'autoscale_on_update': False},
                                     errors=True)
+            self.plot_figure.gca().set_xlim(left=0.0, right=15.0)
+            self.autoscale_y_to_data_in_view()
 
         self.plot_figure.canvas.set_window_title('Muon Analysis')
         self.plot_figure.gca().set_title(title)
 
         self.plot_figure.canvas.window().closing.connect(self._close_plot)
 
+        workspaces_to_remove = [workspace for workspace in self.plotted_workspaces if workspace not in workspace_list]
+        for workspace in workspaces_to_remove:
```

Comment:
can we change this to
```
for workspace in self.plotted_workspaces:
    if workspace not in workspace_list:
        self.remove_workspace_from_plot(workspace)
```
to remove the second loop

Reply:
The remove_workspace_from_plot function modifies the self.plotted_workspaces list removing the relevant work-spaces from it via a list comprehension. Whilst it's possible to remove items from a list in one pass for the size of the lists we are iterating over here I don't think performance is enough of a concern to warrant the increase in complexity.